### PR TITLE
Feedback refactor

### PR DESCRIPTION
**Pull request by lpenzey**
Along with some more minor changes, this pr moves the `main()` entry point from the server to the application and adds tests to the `ServerRunner` class.

### CODE REVIEW

**Comment by lpenzey** [comment]
[Server - Respond to mini review-board feedback - 2](https://trello.com/c/AeJqIY7r/99-server-respond-to-mini-review-board-feedback-2)

File: src/main/java/server/Server.java [hunk 1]
```diff
@@ -1,44 +1,27 @@
...
-                server.start();
-            } catch (ParameterException | IOException e) {
-                System.out.println(e.getMessage());
-            }
-    }
-
-    private void start() throws IOException {
+    public void start() throws IOException {
         ServerSocket serverSocket = new ServerSocket(port);
+        System.out.println("Listening for connection...");
```
Comment (rmulhol) [inline src/main/java/server/Server.java]: could be cool to add the port number to this log

File: src/main/java/server/ServerRunner.java [hunk 2]
```diff
@@ -27,12 +26,18 @@ public void run() {
...
+            if(!input.isEmpty()) {
+                System.out.println("Client request received on "+ Thread.currentThread().getName());
+
                 Request request = requestParser.create(input);
                 Response response = router.generateResponse(request);
                 ResponseFormatter formatter = new ResponseFormatter(response);
 
                 out.print(formatter.stringifyResponse());
+
+                System.out.println("Response sent\n");
```
Comment (rmulhol) [inline src/main/java/server/ServerRunner.java:41]: maybe don't need the trailing `\n` if you're using `println`

File: src/main/java/server/Client.java [hunk 3]
```diff
@@ -21,10 +21,14 @@ public Client(Socket clientSocket) throws IOException {
     public PrintWriter getWriter() { return writer; }
 
     public String chunkStream() {
-        char[] charBuffer = new char[100000];
+        char[] charBuffer = new char[1000];
```
Comment (rmulhol) [inline src/main/java/server/Client.java:24]: What's the effect of making this number smaller?

File: src/main/java/server/Server.java [hunk 4]
```diff
@@ -1,44 +1,27 @@
...
-
 import java.io.IOException;
 import java.net.ServerSocket;
 import java.net.Socket;
 
 public class Server {
-    private Router router;
+
     private Integer port;
     private boolean serverIsRunning = true;
```
Comment (rmulhol) [inline src/main/java/server/Server.java]: perhaps worth defaulting this to `false` and setting to `true` on `start()`?

File: src/main/java/server/Client.java [hunk 5]
```diff
@@ -21,10 +21,14 @@ public Client(Socket clientSocket) throws IOException {
...
-            return new String(charBuffer, 0, bytes_read);
+            String input = "";
+            while(reader.ready()) {
+                int bytes_read = reader.read(charBuffer);
+                input += new String(charBuffer, 0, bytes_read);
+            }
+            return input;
         } catch (IOException | StringIndexOutOfBoundsException error) {
             ServerLogger.serverLogger.log(Level.WARNING, "Error: " + error);
             return null;
```
Comment (rmulhol) [inline src/main/java/server/Client.java]: What happens in this scenario? Are we returning a 500?

**Review by rmulhol — APPROVED** [review]
Really nice work - these changes look great!

File: src/main/java/server/ServerRunner.java (hunk 2 above)
Comment (lpenzey) [reply src/main/java/server/ServerRunner.java:41]: That was just to create some spacing in the console between request/response cycles so it wasn't one long block of text.

File: src/main/java/server/Client.java (hunk 3 above)
Comment (lpenzey) [reply src/main/java/server/Client.java:24]: I'm pretty sure that with the higher buffer `chunkStream` would hit the end of the incoming stream, and for `BufferedReader.read()` hitting the end of stream returns -1, which is what was causing that `indexOutOfBounds` exception to occur.

File: src/main/java/server/Client.java (hunk 5 above)
Comment (lpenzey) [reply src/main/java/server/Client.java]: We are now!

**Pull request merged**